test/unit/base_cli_test.rb

```diff
@@ -0,0 +1,31 @@
+# encoding: utf-8
+# copyright: 2017, Chef Software Inc.
+
+require 'helper'
+require 'thor'
+
+describe 'BaseCLI' do
+  let(:cli) { Inspec::BaseCLI.new }
+
+  describe 'merge_options' do
+    it 'json options override cli' do
+      json = { "backend_cache":true }
+      cli_options = { 'json_config' => 'dummy', 'backend_cache' => false }
+      cli.instance_variable_set(:@options, cli_options)
+      cli.expects(:read_config).returns(json)
```

Comment:
Maybe add a `.with` to this so we show how the options plumb all the way to the config reading?

```ruby
cli.expects(:read_config).with('dummy').returns(json)
```